common/meta/api/src/schema_api_impl.rs

```diff
@@ -1499,21 +1498,11 @@ impl<KV: KVApi> SchemaApi for KV {
     }
 }
 
-// Return true if table is out of `DATA_RETENTION_TIME_IN_DAYS option,
-// use DEFAULT_DATA_RETENTION_SECONDS by default.
-fn is_table_out_of_retention_time(table_meta: &TableMeta, now: &DateTime<Utc>) -> bool {
-    if let Some(drop_on) = table_meta.drop_on {
-        return now.timestamp() - drop_on.timestamp() > DEFAULT_DATA_RETENTION_SECONDS;
-    }
-
-    false
-}
-
-// Return true if db is out of `DATA_RETENTION_TIME_IN_DAYS option,
+// Return true if drop time is out of `DATA_RETENTION_TIME_IN_DAYS option,
 // use DEFAULT_DATA_RETENTION_SECONDS by default.
-fn is_db_out_of_retention_time(db_meta: &DatabaseMeta, now: &DateTime<Utc>) -> bool {
-    if let Some(drop_on) = db_meta.drop_on {
-        return now.timestamp() - drop_on.timestamp() > DEFAULT_DATA_RETENTION_SECONDS;
+fn is_drop_time_out_of_retention_time(drop_on: &Option<DateTime<Utc>>, now: i64) -> bool {
```

Comment:
It looks the original `DateTime<Utc>` would be better than a i64 :D

Reply:
why, if input `now:DateTime<Utc>`, every time call this function it will call `timestamp ` again, and this cost can be reduced.

Reply:
Because there is a chance the caller does not know what `now` stands for: in second or millisecond. And the caller may not use a UTC time to generate the timestamp. And maybe there is a day-night shifting issue that has to be considered.

Thus either use `Datetime<Utc>` everywhere or use `timestamp in second` everywhere.